onnxruntime/core/optimizer/qdq_transformer/selectors_actions/shared/utils.cc

```diff
@@ -0,0 +1,147 @@
+                                 std::move(selector));
+}
+
+void RegisterVariadicSelectors(Selectors& qdq_selectors) {
+  /* register selectors for variadic ops */
+  std::unique_ptr<BaseSelector> selector(new QDQ::VariadicSelector());
+  qdq_selectors.RegisterSelector(GetVariadicOpVersionsMap(),
+                                 std::move(selector));
+}
+
+void RegisterConvSelector(Selectors& qdq_selectors) {
+  /* register selector for conv op */
+  std::unique_ptr<BaseSelector> selector(new QDQ::ConvSelector());
+  qdq_selectors.RegisterSelector(GetConvOpVersionsMap(),
+                                 std::move(selector));
+}
+
+void RegisterMatMulSelector(Selectors& qdq_selectors) {
+  /* register selector for matmul op */
+  std::unique_ptr<BaseSelector> selector(new QDQ::MatMulSelector());
+  qdq_selectors.RegisterSelector(GetMatMulOpVersionsMap(),
+                                 std::move(selector));
+}
+
+void SelectorManager::CreateSelectors() {
+  RegisterMiscSelectors(qdq_selectors_);
+  RegisterUnarySelectors(qdq_selectors_);
+  RegisterBinarySelectors(qdq_selectors_);
+  RegisterVariadicSelectors(qdq_selectors_);
+  RegisterConvSelector(qdq_selectors_);
+  RegisterMatMulSelector(qdq_selectors_);
+}
+
+void SelectorManager::InitializeSelectorsMap() {
+  for (const auto& entry : qdq_selectors_.SelectorsSet()) {
+    for (const auto& op_info : entry->op_versions_map) {
+      bool inserted = op_type_to_selectors_map_.insert({op_info.first, &*entry}).second;
+      ORT_ENFORCE(inserted, "Multiple entries for operator is not supported. OpType=", op_info.first);
+    }
+  }
+}
+
+void SelectorManager::Initialize() {
+  CreateSelectors();
+  InitializeSelectorsMap();
+}
+
+std::vector<NodeGroup> SelectorManager::GetQDQSelections(const GraphViewer& graph_viewer) const {
+  std::vector<NodeGroup> qdq_selections;
+  for (auto index : graph_viewer.GetNodesInTopologicalOrder()) {
+    const auto* node = graph_viewer.GetNode(index);
+    if (node->Domain() != kOnnxDomain) {
+      continue;
+    }
+
+    auto op_rule = op_type_to_selectors_map_.find(node->OpType());
+    if (op_rule == op_type_to_selectors_map_.cend()) {
+      continue;
+    }
+
+    const auto& op_versions_and_selector = *op_rule->second;
+
+    // check the supported versions if specified
+    const auto& versions = op_versions_and_selector.op_versions_map.find(node->OpType())->second;
+    if (!versions.empty()) {
+      if (std::find(versions.cbegin(), versions.cend(), node->SinceVersion()) == versions.cend()) {
+        LOGS_DEFAULT(VERBOSE) << "Op version is not supported for" << node->OpType();
+        continue;
+      }
+    }
+
+    const auto qdq_node_group_selection = op_versions_and_selector.selector->GetQDQSelection(graph_viewer, *node);
+    if (!qdq_node_group_selection.has_value()) {
+      continue;
+    }
+    const auto& qdq_group = *qdq_node_group_selection;
+    qdq_selections.push_back(qdq_group);
+  }
```

Comment:
nit: can avoid a 'continue' given this is the last check. generally the fewer control flow jumps the easier to read

    if (qdq_node_group_selection.has_value()) {
      const auto& qdq_group = *qdq_node_group_selection;
      qdq_selections.push_back(qdq_group);
    }